src/SharedWeb/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -68,41 +68,7 @@ public static class ServiceCollectionExtensions
 {
     public static SupportedDatabaseProviders AddDatabaseRepositories(this IServiceCollection services, GlobalSettings globalSettings)
     {
-        var selectedDatabaseProvider = globalSettings.DatabaseProvider;
-        var provider = SupportedDatabaseProviders.SqlServer;
-        var connectionString = string.Empty;
-
-        if (!string.IsNullOrWhiteSpace(selectedDatabaseProvider))
-        {
-            switch (selectedDatabaseProvider.ToLowerInvariant())
-            {
-                case "postgres":
-                case "postgresql":
-                    provider = SupportedDatabaseProviders.Postgres;
-                    connectionString = globalSettings.PostgreSql.ConnectionString;
-                    break;
-                case "mysql":
-                case "mariadb":
-                    provider = SupportedDatabaseProviders.MySql;
-                    connectionString = globalSettings.MySql.ConnectionString;
-                    break;
-                case "sqlite":
-                    provider = SupportedDatabaseProviders.Sqlite;
-                    connectionString = globalSettings.Sqlite.ConnectionString;
-                    break;
-                case "sqlserver":
-                    connectionString = globalSettings.SqlServer.ConnectionString;
-                    break;
-                default:
-                    break;
-            }
-        }
-        else
-        {
-            // Default to attempting to use SqlServer connection string if globalSettings.DatabaseProvider has no value.
-            connectionString = globalSettings.SqlServer.ConnectionString;
-        }
-
+        var (provider, connectionString) = GetDatabaseProvider(globalSettings);
```

Comment:
I needed this logic for determining what database was being used down below for IDistCache registration, so moved this logic into a method that I can reused.